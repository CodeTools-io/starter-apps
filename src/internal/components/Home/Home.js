import React, { useContext } from 'react';
import { groupBy } from 'lodash';
import {
  Box,
  Card,
  Grid,
  Heading,
  Paragraph,
  ResponsiveContext,
} from 'grommet';
import { Link } from 'react-router-dom';
import Socials from 'internal/components/Socials';

function ComponentSection({ components, description, title }) {
  const size = useContext(ResponsiveContext);

  return (
    <Box gap="medium" pad={{ vertical: 'medium' }}>
      <Heading level={2} margin="none">
        {title}
      </Heading>
      <Paragraph margin="none" fill>
        {description}
      </Paragraph>
      <Grid
        columns={size !== 'small' ? 'medium' : '100%'}
        rows={['small']}
        gap="large"
      >
        {components?.map((component) => {
          return (
            <Link
              key={`${component?.data?.category}-${component?.data?.slug}`}
              to={`/${component?.data?.category}/${component?.data?.slug}`}
              style={{ textDecoration: 'none' }}
            >
              <Card background="white" pad="medium" gap="small" fill>
                <Heading level={4} margin="none">
                  {component?.data?.displayName}
                </Heading>
                <Paragraph margin="none">{component?.data?.short}</Paragraph>
              </Card>
            </Link>
          );
        })}
      </Grid>
    </Box>
  );
}

function HomeHeader() {
  return (
    <Box direction="row" justify="between">
      <Box gap="small" flex={false}>
        <Heading level={1} margin="none">
          Starter Apps
        </Heading>
        <Paragraph margin="none" fill>
          Grommet based UI solutions for various application types.
        </Paragraph>
        <Socials margin={{ top: 'small' }} />
      </Box>
    </Box>
  );
}
export default function Home({ site, ...props }) {
  const docs = groupBy(site?.docs, 'data.category');

  return (
    <Box className="Home" gap="large" flex={false} fill="horizontal" {...props}>
      <HomeHeader />
      <Box gap="large">
        <ComponentSection
          title="Features"
          description="Components that implement a complete product feature in the application."
          components={docs?.apps}
        />

        <ComponentSection
          title="Shells"
          description="Components that wrap an application globally."
          components={docs?.shells}
        />
      </Box>
    </Box>
  );
}
