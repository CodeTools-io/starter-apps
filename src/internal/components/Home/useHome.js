import { useMemo, useState } from 'react';
import { orderBy, uniqBy } from 'lodash';

export default ({ modules = [], components = [], categories = [] }) => {
  const uniqueGrommets = new Set(
    components?.flatMap((c) => c?.grommet)?.filter((c) => c)
  );
  const [grommetOptions, setGrommetOptions] = useState(
    Object.fromEntries([...uniqueGrommets]?.map((c) => [c, false]))
  );
  const [moduleOptions, setModuleOptions] = useState(
    Object.fromEntries(modules?.map((m) => [m?.name, false]))
  );
  const [categoryOptions, setCategoryOptions] = useState(
    Object.fromEntries(categories?.map((c) => [c?.name, false]))
  );

  const filteredCategories = useMemo(() => {
    if (Object.values(categoryOptions)?.every((c) => !c)) {
      return categories;
    }

    return categories?.filter((c) => {
      return categoryOptions[c?.name];
    });
  }, [categories, categoryOptions]);

  const filteredComponents = useMemo(() => {
    const allGrommetUnchecked = Object.values(grommetOptions)?.every((c) => !c);
    const allModulesUnchecked = Object.values(moduleOptions)?.every((c) => !c);
    const allUnchecked = allGrommetUnchecked && allModulesUnchecked;

    if (allUnchecked) {
      return orderBy(components, ['name'], ['asc']);
    }

    const filteredByModules = components?.filter(
      (c) => moduleOptions[c?.data?.module?.name]
    );
    const grommetFilters = Object.entries(grommetOptions)
      ?.filter(([key, val]) => val)
      ?.map(([key, val]) => key);
    const filteredByGrommet = components?.filter((c) =>
      c?.grommet?.some((g) => grommetFilters?.includes(g))
    );

    return orderBy(
      uniqBy([...filteredByModules, ...filteredByGrommet], 'id'),
      ['name'],
      ['asc']
    );
  }, [components, grommetOptions, moduleOptions]);

  return useMemo(() => {
    return {
      categories,
      modules,
      components,
      filteredCategories,
      filteredComponents,
      categoryOptions,
      moduleOptions,
      grommetOptions,
      setCategoryOptions,
      setGrommetOptions,
      setModuleOptions,
    };
  }, [
    categories,
    modules,
    components,
    filteredCategories,
    filteredComponents,
    categoryOptions,
    moduleOptions,
    grommetOptions,
    setCategoryOptions,
    setGrommetOptions,
    setModuleOptions,
  ]);
};
