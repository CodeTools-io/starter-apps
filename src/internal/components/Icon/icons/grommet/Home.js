
import React from 'react';
import { Home } from 'grommet-icons';

export default function HomeIcon({...props}) {
  return <Home {...props}/>
}
  