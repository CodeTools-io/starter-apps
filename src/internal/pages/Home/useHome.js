import { useMemo, useState } from 'react';
import { countBy, orderBy, intersectionBy } from 'lodash';

export default function useHome({
  modules = [],
  components = [],
  categories = [],
}) {
  const uniqueGrommets = useMemo(() => {
    const grommets = new Set(
      components?.flatMap((c) => c?.grommet)?.filter((c) => c)
    );
    return [...grommets]?.sort((a, b) => {
      const nameA = a?.toUpperCase?.();
      const nameB = b?.toUpperCase?.();
      if (nameA < nameB) {
        return -1;
      }
      if (nameA > nameB) {
        return 1;
      }

      return 0;
    });
  }, [components]);
  const initialGrommetOptions = Object.fromEntries(
    [...uniqueGrommets]?.map((c) => [c, false])
  );
  const initialModuleOptions = Object.fromEntries(
    modules?.map((m) => [m?.name, false])
  );
  const initialCategoryOptions = Object.fromEntries(
    categories?.map((c) => [c?.name, false])
  );
  const [grommetOptions, setGrommetOptions] = useState({
    ...initialGrommetOptions,
  });
  const [moduleOptions, setModuleOptions] = useState({
    ...initialModuleOptions,
  });
  const [categoryOptions, setCategoryOptions] = useState({
    ...initialCategoryOptions,
  });

  const filteredCategories = useMemo(() => {
    if (Object.values(categoryOptions)?.every((c) => !c)) {
      return categories;
    }

    return categories?.filter((c) => {
      return categoryOptions[c?.name];
    });
  }, [categories, categoryOptions]);

  const filteredComponents = useMemo(() => {
    const allGrommetUnchecked = Object.values(grommetOptions)?.every((c) => !c);
    const allModulesUnchecked = Object.values(moduleOptions)?.every((c) => !c);
    const allUnchecked = allGrommetUnchecked && allModulesUnchecked;

    if (allUnchecked) {
      return orderBy(components, ['name'], ['asc']);
    }

    const filteredByModules = allModulesUnchecked
      ? components
      : components?.filter(
          (component) => moduleOptions[component?.data?.module?.name]
        );
    const grommetFilters = Object.entries(grommetOptions)
      ?.filter(([key, val]) => val)
      ?.map(([key, val]) => key);
    const filteredByGrommet = allGrommetUnchecked
      ? components
      : components?.filter((component) =>
          grommetFilters?.every((filter) =>
            component?.grommet?.includes(filter)
          )
        );

    return orderBy(
      intersectionBy(filteredByModules, filteredByGrommet, 'id'),
      ['name'],
      ['asc']
    );
  }, [components, grommetOptions, moduleOptions]);

  const grommetCounts = useMemo(() => {
    const counts = [...uniqueGrommets]?.reduce((accum, grommetName) => {
      return {
        ...accum,
        [grommetName]: 0,
      };
    }, {});
    components.forEach((component) => {
      if (component?.grommet) {
        component.grommet.forEach((grommetName) => {
          counts[grommetName] = counts[grommetName] + 1;
        });
      }
    });

    return counts;
  }, [components, uniqueGrommets]);

  const moduleCounts = useMemo(() => {
    return countBy(
      components?.filter((c) => c?.moduleId),
      'data.module.name'
    );
  }, [components]);

  return useMemo(() => {
    function clearOptions() {
      setGrommetOptions({ ...initialGrommetOptions });
      setModuleOptions({ ...initialModuleOptions });
      setCategoryOptions({ ...initialCategoryOptions });
    }
    return {
      categories,
      modules,
      components,
      clearOptions,
      filteredCategories,
      filteredComponents,
      categoryOptions,
      moduleCounts,
      moduleOptions,
      grommetCounts,
      grommetOptions,
      setCategoryOptions,
      setGrommetOptions,
      setModuleOptions,
    };
  }, [
    categories,
    modules,
    components,
    filteredCategories,
    filteredComponents,
    categoryOptions,
    initialGrommetOptions,
    initialModuleOptions,
    initialCategoryOptions,
    moduleCounts,
    moduleOptions,
    grommetCounts,
    grommetOptions,
    setCategoryOptions,
    setGrommetOptions,
    setModuleOptions,
  ]);
}
