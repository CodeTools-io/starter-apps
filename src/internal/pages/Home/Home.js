import React, { useContext, useMemo, useRef, useState } from 'react';
import {
  Anchor,
  Box,
  Button,
  Card,
  CheckBox,
  DropButton,
  Grid,
  Heading,
  Paragraph,
  ResponsiveContext,
  Text,
} from 'grommet';
import { Filter } from 'grommet-icons';
import { Link } from 'react-router-dom';
import PageHeader from 'internal/components/PageHeader';

import useHome from './useHome';

function HomeDocLink({
  data,
  description,
  grommet,
  moduleId,
  name,
  path,
  ...props
}) {
  return (
    <Link to={path} style={{ textDecoration: 'none' }}>
      <Card background="white" pad="medium" gap="small" fill>
        <Box gap="xxsmall">
          <Heading level={4} margin="none">
            {name}
          </Heading>
          {moduleId && (
            <Text size="small" color="dark-6">
              {data?.module?.name}
            </Text>
          )}
        </Box>
        <Paragraph margin="none">{description}</Paragraph>
      </Card>
    </Link>
  );
}

function HomeSection({ components, description, title, ...props }) {
  const size = useContext(ResponsiveContext);

  return (
    <Box gap="medium" pad={{ vertical: 'medium' }} {...props}>
      <Heading level={2} margin="none">
        {title}
      </Heading>
      <Paragraph margin="none" fill>
        {description}
      </Paragraph>
      <Grid
        columns={size !== 'small' ? 'medium' : '100%'}
        rows={['small']}
        gap="large"
      >
        {components?.map((component) => (
          <HomeDocLink key={component?.path} {...component} />
        ))}
      </Grid>
    </Box>
  );
}

function HomeFiltersMenu({ sections, onClear }) {
  const size = useContext(ResponsiveContext);
  const showClear = useMemo(() => {
    return sections?.some((section) => {
      return Object.values(section?.options)?.some((option) => option);
    });
  }, [sections]);
  return (
    <Box
      margin={{ top: 'small' }}
      gap="large"
      pad="medium"
      round="small"
      background="white"
      elevation="small"
      flex={false}
      width={size === 'small' ? '90vw' : '100%'}
    >
      {sections?.map((section) => {
        return (
          <Box key={section?.key} gap="small" flex={false}>
            <Heading margin="none" level={4}>
              {section?.title}
            </Heading>
            <Grid gap="small" columns="small" rows={['auto']} fill>
              {Object.keys(section?.options)?.map((option) => {
                const labelSuffix = section?.counts
                  ? ` (${section?.counts?.[option]})`
                  : null;
                return (
                  <Box key={`${section?.key}-filter-${option}`}>
                    <CheckBox
                      tabIndex="-1"
                      label={
                        <Text size="small">
                          {option}
                          {labelSuffix}
                        </Text>
                      }
                      checked={section?.options?.[option]}
                      onChange={(event) =>
                        section?.updateOptions({
                          ...section?.options,
                          [option]: event?.target?.checked,
                        })
                      }
                    />
                  </Box>
                );
              })}
            </Grid>
          </Box>
        );
      })}
      {showClear && (
        <Box direction="row">
          <Button label="Clear" onClick={onClear} color="control" plain />
        </Box>
      )}
    </Box>
  );
}

function HomeFilters({ sections = [], onClear = () => {}, ...props }) {
  const size = useContext(ResponsiveContext);
  const filterMenuRef = useRef();
  const [showDrop, setShowDrop] = useState(false);
  const appliedFilters = useMemo(() => {
    return sections
      ?.map((section) => {
        const title = section?.title;
        const filters = Object.entries(section?.options)
          ?.filter(([key, val]) => val)
          ?.map(([key]) => key);
        return { title, filters };
      })
      ?.filter((section) => section?.filters?.length);
  }, [sections]);
  const showAppliedFilters = useMemo(() => {
    if (!showDrop && appliedFilters?.length) {
      return true;
    }

    return false;
  }, [appliedFilters, showDrop]);

  return (
    <Box fill="horizontal" {...props}>
      <Box direction="row" justify="between" align="baseline" gap="medium">
        <Box direction="row" gap="small">
          {showAppliedFilters && <Text weight="bold">Filtered By: </Text>}
          {showAppliedFilters &&
            appliedFilters?.map((appliedFilters) => {
              return (
                <Box
                  key={`applied-filter-${appliedFilters?.title}`}
                  direction="row"
                  align="baseline"
                  gap="small"
                >
                  {appliedFilters?.filters?.map((filter) => {
                    return (
                      <Text
                        key={`applied-filter-${appliedFilters?.title}-${filter}`}
                      >
                        {filter}
                      </Text>
                    );
                  })}
                </Box>
              );
            })}
          {showAppliedFilters && (
            <Button label="Clear" onClick={onClear} color="control" plain />
          )}
        </Box>
        <DropButton
          label="filter"
          icon={<Filter size="small" />}
          onOpen={() => setShowDrop(!showDrop)}
          onClose={() => setShowDrop(!showDrop)}
          dropAlign={{ top: 'bottom', right: 'right' }}
          dropContent={
            <HomeFiltersMenu sections={sections} onClear={onClear} />
          }
          dropProps={{
            overflow: 'visible',
            plain: true,
          }}
          dropTarget={size !== 'small' ? filterMenuRef.current : null}
        />
      </Box>
      <Box
        fill="horizontal"
        style={{ position: size !== 'small' ? 'relative' : 'static' }}
      >
        <Box
          ref={filterMenuRef}
          style={{ position: 'absolute', top: '100%' }}
          fill="horizontal"
        ></Box>
      </Box>
    </Box>
  );
}

export default function Home({ docs = {}, ...props }) {
  const {
    categories,
    clearOptions,
    filteredComponents,
    moduleCounts,
    moduleOptions,
    grommetOptions,
    grommetCounts,
    setGrommetOptions,
    setModuleOptions,
  } = useHome({ ...docs });

  return (
    <Box className="Home" flex={false} fill="horizontal" {...props}>
      <PageHeader
        title="Starter Apps"
        description={
          <Text>
            Examples of React based applications built with{' '}
            <Anchor href="https://v2.grommet.io" target="_blank">
              Grommet
            </Anchor>
          </Text>
        }
        margin={{ top: 'small', bottom: 'large' }}
      />

      <HomeFilters
        onClear={clearOptions}
        sections={[
          {
            key: 'module-filters',
            title: 'Modules',
            options: moduleOptions,
            updateOptions: setModuleOptions,
            counts: moduleCounts,
          },
          {
            key: 'grommet-filters',
            title: 'Components',
            options: grommetOptions,
            updateOptions: setGrommetOptions,
            counts: grommetCounts,
          },
        ]}
      />

      <Box gap="large">
        {categories?.map((category) => {
          return (
            <HomeSection
              key={category.data.path}
              title={category.data.name}
              description={category.data.description}
              components={filteredComponents.filter(
                (component) => component.data.categoryId === category.data.id
              )}
            />
          );
        })}
      </Box>
    </Box>
  );
}
