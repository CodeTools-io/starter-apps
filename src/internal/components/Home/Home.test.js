import React from 'react'
import Home from './Home'

describe('Home', () => {
  test('can render', () => {
    const { container } = renderComponent(
      <Home>
        <p>Content for the Home</p>
      </Home>
    )

    expect(container).toMatchSnapshot()
  })
})

