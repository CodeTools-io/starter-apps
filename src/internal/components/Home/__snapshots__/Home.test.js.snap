// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home can render 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 kYvUof Home"
  >
    <div
      class="StyledBox-sc-13pk1d4-0 dlZDBx"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 gzfjUM"
      >
        <h1
          class="StyledHeading-sc-1rdh4aw-0 bGlFVF"
        >
          Starter Apps
        </h1>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
        />
        <p
          class="StyledParagraph-tbetod-0 dTCtkV"
        >
          Grommet based UI solutions for various application types.
        </p>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
        />
        <div
          class="StyledBox-sc-13pk1d4-0 kiGBNl"
        >
          <a
            class="StyledAnchor-sc-1rp7lwl-0 dajpoL"
            href="https://github.com/<github_user>/<github_repo_name>"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 rpWqh"
            >
              <svg
                aria-label="Github"
                class="StyledIcon-ofa7kd-0 cQVwrT"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11.9989871,1 C5.92550416,1 1,5.92482888 1,12.0003376 C1,16.8603395 4.15153934,20.9829338 8.52263728,22.4374904 C9.0729918,22.5387827 9.27355045,22.199116 9.27355045,21.9073943 C9.27355045,21.6467356 9.2640965,20.954572 9.25869425,20.0368642 C6.19899322,20.7013414 5.55342398,18.5620492 5.55342398,18.5620492 C5.0530403,17.2911692 4.33183953,16.9528531 4.33183953,16.9528531 C3.33309801,16.2708186 4.40747107,16.2843243 4.40747107,16.2843243 C5.51155652,16.3619816 6.09229872,17.4181221 6.09229872,17.4181221 C7.07348292,19.0988981 8.66714755,18.6133706 9.2938089,18.3317781 C9.39375058,17.6213819 9.67804414,17.1365297 9.99205009,16.86169 C7.54955646,16.5841493 4.98146045,15.6401056 4.98146045,11.4249977 C4.98146045,10.224347 5.41026428,9.24181221 6.11390773,8.47334172 C6.00046042,8.19512569 5.62297799,7.07618404 6.22195279,5.56220265 C6.22195279,5.56220265 7.14506277,5.26642929 9.24653918,6.68992296 C10.12373,6.44547101 11.0650726,6.32392032 12.0003376,6.31919335 C12.9349274,6.32392032 13.8755947,6.44547101 14.7541361,6.68992296 C16.8542619,5.26642929 17.7760214,5.56220265 17.7760214,5.56220265 C18.3763467,7.07618404 17.9988643,8.19512569 17.8860923,8.47334172 C18.5910863,9.24181221 19.0165137,10.224347 19.0165137,11.4249977 C19.0165137,15.6509101 16.444366,16.5807729 13.9944443,16.8529114 C14.3888087,17.192578 14.7406305,17.863808 14.7406305,18.890236 C14.7406305,20.3603241 14.7271248,21.5467939 14.7271248,21.9073943 C14.7271248,22.2018171 14.9256576,22.5441849 15.4834403,22.4368151 C19.8511618,20.9788821 23,16.8589889 23,12.0003376 C23,5.92482888 18.0744958,1 11.9989871,1"
                  fill="#333"
                  fill-rule="evenodd"
                />
              </svg>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 ffjnxm"
              />
              <span
                class="StyledText-sc-1sadyjn-0 hncrgS"
              >
                View Source
              </span>
            </div>
          </a>
        </div>
      </div>
    </div>
    <div
      class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDebtf"
    />
    <div
      class="StyledBox-sc-13pk1d4-0 bFwQzJ"
    >
      <div
        class="StyledBox-sc-13pk1d4-0 jPjyPf"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 jXxrrh"
        >
          Features
        </h2>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
        />
        <p
          class="StyledParagraph-tbetod-0 dTCtkV"
        >
          Components that implement a complete product feature in the application.
        </p>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
        />
        <div
          class="StyledGrid-sc-1wofa1l-0 jHTOBO"
        >
          <a
            href="/apps/calendar"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Calendar
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for managing dates, events, and reminders.
              </p>
            </div>
          </a>
          <a
            href="/apps/commerce/cart"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Cart
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for showing items that a customer intends to purchase.
              </p>
            </div>
          </a>
          <a
            href="/apps/chat"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Chat
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for facilitating real-time text communication.
              </p>
            </div>
          </a>
          <a
            href="/apps/commerce/checkout"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Checkout
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for showing items that a customer is about to purchase.
              </p>
            </div>
          </a>
          <a
            href="/apps/contacts"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Contacts
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for managing information about lists of people.
              </p>
            </div>
          </a>
          <a
            href="/apps/dashboard"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Dashboard
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for viewing high-level information about systems and organizations.
              </p>
            </div>
          </a>
          <a
            href="/apps/email"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Email
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for managing email.
              </p>
            </div>
          </a>
          <a
            href="/apps/social-media/feed"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Feed
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for displaying a stream of live social media posts
              </p>
            </div>
          </a>
          <a
            href="/apps/notes"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Notes
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for taking notes.
              </p>
            </div>
          </a>
          <a
            href="/apps/social-media/profile"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Profile
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for showing social information about a person.
              </p>
            </div>
          </a>
          <a
            href="/apps/project-manager"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Project Manager
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for managing team projects.
              </p>
            </div>
          </a>
          <a
            href="/apps/commerce/store"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                Store
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A feature for showing products to a customer.
              </p>
            </div>
          </a>
        </div>
      </div>
      <div
        class="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDebtf"
      />
      <div
        class="StyledBox-sc-13pk1d4-0 jPjyPf"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 jXxrrh"
        >
          Shells
        </h2>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
        />
        <p
          class="StyledParagraph-tbetod-0 dTCtkV"
        >
          Components that wrap an application globally.
        </p>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 jtjTeE"
        />
        <div
          class="StyledGrid-sc-1wofa1l-0 jHTOBO"
        >
          <a
            href="/shells/app-shell"
            style="text-decoration: none;"
          >
            <div
              class="StyledBox-sc-13pk1d4-0 bDlKMv"
            >
              <h4
                class="StyledHeading-sc-1rdh4aw-0 bqNAXo"
              >
                App Shell
              </h4>
              <div
                class="StyledBox__StyledBoxGap-sc-13pk1d4-1 liuKJs"
              />
              <p
                class="StyledParagraph-tbetod-0 hFElNu"
              >
                A wrapper with a logo, navigation, router, notification system, search, and user menu.
              </p>
            </div>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
